core/leader/leader.go

```diff
@@ -13,18 +13,29 @@ import (
 	"chain/log"
 )
 
-var (
-	isLeading bool
-	lock      sync.Mutex
-)
+type Leadership interface {
+	IsLeading() bool
+	Address(ctx context.Context) (string, error)
+}
 
-// IsLeading returns true if this process is
-// the core leader.
-func IsLeading() bool {
-	lock.Lock()
-	l := isLeading
-	lock.Unlock()
-	return l
+type leadership struct {
```

Comment:
> Do we need the interface? It would be more conventional to export this type.

The main motivation for this PR was for an interface so that this test could mock the leadership: https://github.com/chain/chain/pull/103/files#diff-319ba1ae4ee007599fec4654b2e71b6cR224

Not sure if there's a better alternative.